File: Mavim-services/Topic/Mavim.Manager.Api.Topic.Repository/v1/Mappers/MultiNumberFieldMapper.cs
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.enums;
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Mappers.Abstract;
using System;
using System.Linq;

namespace Mavim.Manager.Api.Topic.Repository.v1.Mappers
{
    internal class MultiNumberFieldMapper : GenericNumberFieldMapper<Model.IMultiValueSimpleField, IMultiNumberField>
    {
        /// <summary>
        /// Gets the generic mapped number service field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <returns></returns>
        /// <exception cref="ArgumentNullException">field</exception>
        protected override IMultiNumberField GetGenericMappedServiceField(Model.IMultiValueSimpleField field)
        {
            if (field == null) throw new ArgumentNullException(nameof(field));

            return new MultiNumberField
            {
                FieldId = field.FieldDefinition.ID,
                FieldName = field.FieldDefinition.Name,
                SetOrder = field.FieldSetDefinition.OrderNumber,
                Order = field.OrderNumber,
                FieldSetId = field.FieldSetDefinition.ID,
                SetName = field.FieldSetDefinition?.Name ?? string.Empty,
                FieldValues = field.FieldValues?.Select(i => i?.ToString()),
                FieldValueType = FieldType.MultiNumber,
                Required = field.FieldDefinition.Required,
                Readonly = field.ReadOnly
            };
        }

        /// <summary>
        /// Gets the generic mapped number repo field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <param name="simpleField">The simple field.</param>
        /// <returns></returns>
        /// <exception cref="ArgumentNullException">field</exception>
        protected override object[] GetGenericMappedRepoField(IMultiNumberField field, Model.ISimpleField simpleField = null)
        {
            if (field == null) return new object[] { null };

            return field.FieldValues == null ? new object[] { null } : field.FieldValues.ToArray<object>();
        }
    }
}

File: Mavim-apps/Mavim.frontend/projects/mavim-app/src/shared/field/mappers/multinumberfield.mapper.ts
import { Field } from '../models/field.model';
import { MultiNumberField } from '../models/api';
import { GenericNumberFieldMapper } from './abstract/generic-number-field.mapper';
import { mapToApiFieldType, mapToFieldType } from './fieldvaluetype';

export class MultiNumberFieldMapper extends GenericNumberFieldMapper<MultiNumberField> {
	public get getUrlTypeName(): string {
		return 'multinumber';
	}

	public getMappedField(field: MultiNumberField): Field {
		return this.mapField(field);
	}

	public getMappedApiField(field: Field): MultiNumberField {
		return this.mapApiField(field);
	}

	protected mapApiField(field: Field): MultiNumberField {
		const data = field.data
			? [...field.data].map((d) => this.mapToNumber(d))
			: undefined;
		return {
			...field,
			fieldValueType: mapToApiFieldType(field.fieldValueType),
			data
		} as MultiNumberField;
	}

	protected mapField(field: MultiNumberField): Field {
		return {
			...field,
			fieldValueType: mapToFieldType(field.fieldValueType),
			options: undefined,
			data: field.data
				? [...field.data].map((d) => this.mapToString(d))
				: []
		} as Field;
	}
}
